base/twiceprecision.jl

```diff
@@ -638,25 +648,26 @@ function range_start_stop_length(start::T, stop::T, len::Integer) where {T<:IEEE
 end
 
 function _linspace(start::T, stop::T, len::Integer) where {T<:IEEEFloat}
+    len = len + 0 # promote with Int
     (isfinite(start) && isfinite(stop)) || throw(ArgumentError("start and stop must be finite, got $start and $stop"))
     # Find the index that returns the smallest-magnitude element
     Δ, Δfac = stop-start, 1
     if !isfinite(Δ)   # handle overflow for large endpoints
-        Δ, Δfac = stop/len - start/len, Int(len)
+        Δ, Δfac = stop/len - start/len, len
     end
     tmin = -(start/Δ)/Δfac            # t such that (1-t)*start + t*stop == 0
-    imin = round(Int, tmin*(len-1)+1) # index approximately corresponding to t
+    imin = round(typeof(len), tmin*(len-1)+1) # index approximately corresponding to t
     if 1 < imin < len
         # The smallest-magnitude element is in the interior
         t = (imin-1)/(len-1)
         ref = T((1-t)*start + t*stop)
         step = imin-1 < len-imin ? (ref-start)/(imin-1) : (stop-ref)/(len-imin)
     elseif imin <= 1
-        imin = 1
+        imin = oneunit(typeof(len))
         ref = start
         step = (Δ/(len-1))*Δfac
     else
-        imin = Int(len)
+        imin = len
```

Comment:
The `2` and `1` on the call to `steprangelen_hp` below might also need their types adjusted.